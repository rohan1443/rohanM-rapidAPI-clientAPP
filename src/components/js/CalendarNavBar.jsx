import React, { Component } from 'react'
import { connect } from 'react-redux'

import { selectRoute } from '../../actions'
import { COUNTRY_CODES } from '../../constants'

export class CalendarNavBar extends Component {
  state = {
    
  }

  generateRouteLabel = (source, destination) => {
    return `${COUNTRY_CODES[source]} - ${COUNTRY_CODES[destination]}`
  }
  generateRoutes = (source, destination) => {
    return { source, destination }
  }

  render() {
    const { calendarLoading } = this.props

    return (
      <div className="calendar-navbar-container">
        <button disabled={calendarLoading} className="flight-search" onClick={() => this.props.selectRoute(this.generateRoutes("Singapore", "Kualalumpur"))}>
          {this.generateRouteLabel("Singapore", "Kualalumpur")}
        </button>
        <button disabled={calendarLoading} className="flight-search" onClick={() => this.props.selectRoute(this.generateRoutes("Kualalumpur", "Singapore"))}>
          {this.generateRouteLabel("Kualalumpur", "Singapore")}
        </button>
        <button disabled={calendarLoading} className="flight-search" onClick={() => this.props.selectRoute(this.generateRoutes("Kualalumpur", "SanFransisco"))}>
          {this.generateRouteLabel("Kualalumpur", "SanFransisco")}
        </button>
      </div>
    )
  }
}

const mapStateToProps = ({ calendarLoading }) => {
  return { calendarLoading }
}

export default connect(mapStateToProps, { selectRoute })(CalendarNavBar)