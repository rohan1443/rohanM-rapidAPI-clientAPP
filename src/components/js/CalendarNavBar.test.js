import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

import { CalendarNavBar } from './CalendarNavBar'

describe("Unit testing for Calendar Nav unconnected component", () => {
  
  test("snapshot test for the CalendarNavBar component", () => {
    const wrapper = renderer.create(<CalendarNavBar />).toJSON

    expect(wrapper).toMatchSnapshot()
  })

})