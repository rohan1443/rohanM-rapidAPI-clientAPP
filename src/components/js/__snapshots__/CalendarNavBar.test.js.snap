// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing for Calendar Nav unconnected component snapshot test for the CalendarNavBar component 1`] = `[Function]`;
